### ISSUE TITLE
False negative with needless_raw_string_hashes

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/rust-clippy#11420]
### Summary

The `needless_raw_string_hashes` warning isn't warning me about some cases I thought it would.

### Lint Name

needless_raw_string_hashes

### Reproducer

I tried this code:

```rust
let _y = r###"wea ther"###;
```

I expected to see this happen:

I expected it to tell me that three hashes were unnecessary, and that it could actually be no hashes, and suggest:

```rust
let _y = r"wea ther";
```

which also compiles.

Instead, this happened:

Clippy did not give me any warnings.

### Version

```text
rustc 1.72.0 (5680fa18f 2023-08-23)
binary: rustc
commit-hash: 5680fa18feaa87f3ff04063800aec256c3d4b4be
commit-date: 2023-08-23
host: aarch64-apple-darwin
release: 1.72.0
LLVM version: 16.0.5
```

**Comment by carols10cents** [comment]
Other cases I expected Clippy to warn me about but it doesn't (these raw strings compile successfully with the hashes removed):

```rust
r#"wea=ther"#
r#"wea,ther"#
r#"999999999999"#;
```

I'm wondering if the commonality with all these cases is that they don't actually need to be raw strings at all, but `needless_raw_strings` is `allow` by default? Is the code checking for `needless_raw_strings` and, if that's true, not checking for `needless_raw_string_hashes`, perhaps?

**Comment by carols10cents** [comment]
Kinda looks like that might be it because of this `return`? https://github.com/rust-lang/rust-clippy/blob/4118738998993f0c4b0eb30eda502f21ebd28c4e/clippy_lints/src/raw_strings.rs#L89

**Comment by Centri3** [comment]
Yep, that's it. We should only skip linting if `needless_raw_strings` is warn/deny.
